test/distributed/fsdp/test_fsdp_unshard_params.py

```diff
@@ -228,6 +228,26 @@ def test_unshard_params_param_data(self):
             self._test_unshard_params_param_data,
         )
 
+    @skip_if_lt_x_gpu(2)
+    def test_unshard_params_multiple_modules(self):
```

Comment:
Computing the expected numel as the product of the sharded parameter size and the world size is only true in this no padding case, which could be brittle (e.g. if the sharding changes). Should we consider comparing the result of nested calls to `summon_full_params()` each on a single module vs. a single call to `summon_full_params()` on multiple modules at once? (This is what I interpret as the direct semantics of generalizing the API.)

I am also curious if this is testing the expected use case for adding this feature. When does a user have multiple distinct FSDP modules? I was thinking about the case where we want to use `summon_full_params(recurse=False)` on multiple modules belonging to the same module tree.

Reply:
> like testing use cases when we want to run this for multiple FSDP modules and
compare things.

Does this imply that this feature is meant for testing?

Reply:
@awgu I was also thinking of the use case of 2 separate encoders both wrapped with FSDP, and users want to call summon_full_params on both of them.